quadriga/src/main/java/edu/asu/diging/quadriga/aspect/AuthAspect.java

```diff
@@ -0,0 +1,52 @@
+package edu.asu.diging.quadriga.aspect;
+
+import org.aspectj.lang.ProceedingJoinPoint;
+import org.aspectj.lang.annotation.Around;
+import org.aspectj.lang.annotation.Aspect;
+import org.aspectj.lang.annotation.Pointcut;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.security.core.Authentication;
+
+import edu.asu.diging.quadriga.config.web.TokenInfo;
+
+@Aspect
+@Configuration
+public class AuthAspect {
+
+    @Pointcut("execution(@edu.asu.diging.quadriga.aspect.annotation.InjectToken * *(..))")
+    public void annotatedMethod() {}
+
+    @Pointcut("execution(* *(.., org.springframework.security.core.Authentication, ..))")
+    public void authenticationParamMethod() {}
+
+    @Pointcut("execution(* *(.., edu.asu.diging.quadriga.config.web.TokenInfo, ..))")
+    public void tokenParamMethod() {}
+
+    @Around("annotatedMethod() && authenticationParamMethod() && tokenParamMethod()")
+    public Object authAdvice(ProceedingJoinPoint joinPoint) throws Throwable {
+        Object[] args = joinPoint.getArgs();
+        int authIndex = getIndexOf(args, Authentication.class);
+        int tokenIndex = getIndexOf(args, TokenInfo.class);
+
+        if (authIndex >= 0 && tokenIndex >= 0) {
```

Comment:
if you don't use indexes (see comment below) you simply check for null here.

Reply:
I need the tokenIndex to replace the existing object in args array